lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+ * @param {object} streamingConfig.config - A `RecognitionConfig` object. See
+ *     [`RecognitionConfig`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig).
+ * @param {string} streamingConfig.config.encoding - Encoding of audio data sent
+ *     in all `RecognitionAudio` messages. See
+ *     [`RecognitionConfig.encoding`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {number} streamingConfig.config.sampleRate - Sample rate in Hertz of
+ *     the audio data sent in all `RecognitionAudio` messages. See
+ *     [`RecognitionConfig.sampleRate`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {string=} streamingConfig.config.languageCode - See
+ *     [`RecognitionConfig.languageCode`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `"en-US"`.
+ * @param {number=} streamingConfig.config.maxAlternatives - See
+ *     [`RecognitionConfig.maxAlternatives`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `1`.
+ * @param {boolean=} streamingConfig.config.profanityFilter - See
+ *     [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=} streamingConfig.config.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {boolean=} streamingConfig.singleUtterance - See
+ *     [`StreamingRecognitionConfig.singleUtterance`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {boolean=} streamingConfig.interimResults - See
+ *     [`StreamingRecognitionConfig.interimResults`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @returns {object} Readable and writable stream.
+ *
+ * @example
+ * var fs = require('fs');
+ *
+ * // See the link in the parameters for `RecognizeRequest`.
+ * var streamingConfig = {
+ *   config: {
+ *     encoding: 'LINEAR16',
+ *     sampleRate: 16000
+ *   },
+ *   singleUtterance: false,
+ *   interimResults: false
+ * };
+ *
+ * var recognizeStream = speech.streamingDetect(streamingConfig)
+ *   .on('data', function (speechRecognitionAlternatives) {
```

Comment:
I could be wrong, but is this meant to be singular? (`speechRecognitionAlternative`)